src/main/java/frc/robot/commands/ArmCommand.java

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import com.revrobotics.AbsoluteEncoder;
+import com.revrobotics.SparkMaxAbsoluteEncoder;
+import com.revrobotics.CANSparkMax.ControlType;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.config.Config;
+import frc.robot.subsystems.ArmSubsystem;
+import frc.robot.subsystems.SwerveSubsystem;
+import java.util.Arrays;
+
+public class ArmCommand extends CommandBase {
+  String level;
+  boolean isTop;
+  double integralMinimum = -0.5;
+  double integralMaximum = 0.5;
+  double z;
+  double nodeX;
+  double armSpeed = 0.2;
+  double defaultAngle1 = Math.PI; // angle in radians of joint 1 in default position (vertical)
+  double defaultAngle2 = Math.PI/36; // angle in radians of joint 2 in default position (angle of 5 degrees)
+  double[] defaultAngles;
+  double targetAngularDistanceBottomArm;
+  double targetAngularDistanceTopArm;
+
+  /** Creates a new ArmExtend. */
+  
+  public ArmCommand(String level, boolean isTop) {
+    this.level = level;
+    this.isTop = isTop;
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(ArmSubsystem.getInstance());
+  }
+
+
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    if (level=="1") {
+      z = 4;//constants --> first and second z positions - depends on the height of the node we are going for
+      nodeX = 19.382;//the x positiion of the node we are going for
+    } 
+    else if(level=="2") {
+      z = 24;
+      nodeX = 37.132;
+    }
+    else if (level=="3") {
+      z = 35;
+      nodeX = 54.132;
+    }
+    else if (level=="default") {
+      defaultAngles = new double[]{defaultAngle1, defaultAngle2};
+      z = 0;
+      nodeX = 0;
+    }
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+
+    double x = ArmSubsystem.getInstance().setx(0,nodeX);
+    double[] angles = ArmSubsystem.getInstance().calculateAngle(Config.Arm.L1, Config.Arm.L2, x, z);
+    double angle1 = angles[0];
+    double angle2 = angles[1];
+
+    targetAngularDistanceBottomArm = ArmSubsystem.getInstance().getAngularDistance(angle1, Config.Arm.NEO_GEAR_RATIO);
+    targetAngularDistanceTopArm = ArmSubsystem.getInstance().getAngularDistance(angle2, Config.Arm.NEO_GEAR_RATIO);
+    
+    if (level == "3") {
```

Comment:
Fun fact, you can use something called a switch statement here instead of if-else-if statements. Syntax is:
switch (level)
case "1":
do whatever;
case "2":
...
Functionally it's not much different, but it's quite a bit easier to read and understand what's going on with the switch statements